### PR TITLE
tests(steps): add axe tests, correct steps aria

### PR DESCRIPTION
**Pull request by mmazanec22**
An example of using [Deque's axe library](https://github.com/dequelabs/axe-core) (via [jest-axe](https://www.npmjs.com/package/jest-axe)) to test accessibility for a component (see #596 ). Also fixes an accessibility issue for the steps component. Modeled after #841, assuming that more tests will be added following this pattern.

## Proposed Changes

- Use jest-axe to test accessibility of the Steps component given different props scenarios
- Remove the [aria-orientation](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-orientation) role, as it is not valid on this component
- Add the [aria-current](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-current) property

### CODE REVIEW

File: packages/oruga/package.json [hunk 1]
```diff
@@ -100,5 +100,8 @@
     "vue-component-meta": "^1.8.27",
     "vue-component-type-helpers": "^1.8.27",
     "vue-tsc": "^1.8.27"
+  },
+  "dependencies": {
+    "jest-axe": "^8.0.0"
```
Comment (mlmoravek) [inline packages/oruga/package.json:105]: Is it possile to the dependency as dev dependency?

File: packages/oruga/src/components/steps/tests/StepsExample.vue [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+        default: false,
+    },
+});
+
+const activeStep = ref<number>(1);
+const enableProfileActivateEvent = ref(false);
+
+const onProfileActivate = () => {
+    if (enableProfileActivateEvent.value) {
+        alert("Profile Activated");
```
Comment (mlmoravek) [inline packages/oruga/src/components/steps/tests/StepsExample.vue:51]: I think using alert here seems wrong. I don't know how it works with a virtual dom?

File: packages/oruga/src/components/steps/tests/StepsExample.vue [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+};
+</script>
+
+<template>
+    <!-- Taken from the first example in the documentation -->
+    <o-steps
+        v-model="activeStep"
+        :vertical="isVertical"
+        :animated="isAnimated"
+        :rounded="isRounded"
```
Comment (mlmoravek) [inline packages/oruga/src/components/steps/tests/StepsExample.vue:62]: You only need to specify props for this test component that you will not be using only for the root element. Anything else can be passed to OSteps with `:v-bing=$attrs`.

File: packages/oruga/src/components/steps/tests/steps.integration.test.ts [hunk 4]
```diff
@@ -0,0 +1,59 @@
+import { afterEach, describe, expect, test } from "vitest";
+import { enableAutoUnmount, mount } from "@vue/test-utils";
+import { axe, toHaveNoViolations } from "jest-axe";
+
+import StepsExample from "./StepsExample.vue";
+
+expect.extend(toHaveNoViolations);
```
Comment (mlmoravek) [inline packages/oruga/src/components/steps/tests/steps.integration.test.ts:7]: Do we have to add this in every test file now? Or is it a one time think? If so, you can also add a `vitest.setup.ts` file in a `/src/__tests__/` folder and a reference to the setup file in `vite.config.ts` with `test:{ setupFiles: [require.resolve("./src/__tests__/vitest.setup.ts")],}`

File: packages/oruga/src/components/steps/tests/steps.integration.test.ts [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+        },
+        {
+            title: "axe steps - customNavigation",
+            props: { customNavigation: true },
+        },
+    ];
+
+    test.each(a11yCases)("$title", async ({ props }) => {
+        const wrapper = mount(StepsExample, props);
+        expect(await axe(wrapper.html())).toHaveNoViolations();
```
Comment (mlmoravek) [inline packages/oruga/src/components/steps/tests/steps.integration.test.ts:57]: Wow, thats all? nice :D

File: packages/oruga/src/components/steps/tests/StepsExample.vue (hunk 2 above)
Comment (mmazanec22) [reply packages/oruga/src/components/steps/tests/StepsExample.vue:51]: I copied this from the docs. Happy to remove or replace. I think as long as there is any func here it should satisfy our testing needs

File: packages/oruga/src/components/steps/tests/StepsExample.vue (hunk 3 above)
Comment (mmazanec22) [reply packages/oruga/src/components/steps/tests/StepsExample.vue:62]: @mlmoravek I tried taking these out but realized: I want to toggle one prop at a time to better isolate a11y problems. In order to do that, I want to set default values in props that stay consistent regardless of what the component's defaults are. What do you think about leaving this as is?

Comment (mlmoravek) [reply packages/oruga/src/components/steps/tests/StepsExample.vue:62]: I think for now its fine :) I can refactor it late on when we will add more steps tests. 
To demonstrate how we can benefit from axe tests its good.

**Pull request merged**